service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/service/EncounterReportBundleService.java

```diff
@@ -264,4 +266,25 @@ private void addQuestionnaireResponses(Bundle bundle, List<QuestionnaireResponse
     private void addObservations(Bundle bundle, List<Observation> observations) {
         observations.forEach(observation -> addEntry(bundle, observation));
     }
+
+    private void addRelatedPerson(Bundle bundle, RelatedPerson relatedPerson) {
+        if (relatedPerson != null) {
+            addEntry(bundle, relatedPerson);
+        }
+    }
+
+    private void addHealthcareFacilityLocation(Bundle bundle, Location location, Encounter encounter) {
+        if (location != null) {
+            addEntry(bundle, location);
+
+            Encounter.EncounterLocationComponent encounterLocationComponent = new Encounter.EncounterLocationComponent();
```

Comment:
Can we move EncounterLocationComponent logic to Encounter mapper?

Reply:
Done.